packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -203,3 +257,49 @@ class _BottomAppBarClipper extends CustomClipper<Path> {
         || oldClipper.notchMargin != notchMargin;
   }
 }
+
+class _DefaultsM2 extends BottomAppBarTheme {
+  const _DefaultsM2(this.context)
+    : super(
+      elevation: 8.0,
+    );
+
+  final BuildContext context;
+
+  @override
+  Color? get color => Theme.of(context).bottomAppBarColor;
+
+  @override
+  Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES - BottomAppBar
+
+// Do not edit by hand. The code between the "BEGIN GENERATED" and
+// "END GENERATED" comments are generated from data in the Material
+// Design token database by the script:
+//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Token database version: v0_101
+
+// Generated version v0_101
+class _TokenDefaultsM3 extends BottomAppBarTheme {
+  const _TokenDefaultsM3(this.context)
+    : super(
+      elevation: 3.0,
+      height: 80.0,
+    );
+
+  final BuildContext context;
+
+  @override
+  Color? get color => Theme.of(context).colorScheme.surface;
+
+  @override
+  Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+
+  @override
+  NotchedShape? get shape => const AutomaticNotchedShape(RoundedRectangleBorder());
```

Comment:
This is why we need the template to not add the `const` to the `RoundedRectangleBorder()` shape. Otherwise the next time someone generates the defaults to this will get overridden with a lint warning for the unnecessary `const`.